### ISSUE TITLE
Support tls flag

### ISSUE THREAD

**Issue by arthurk** [bacongobbler/helm-whatup#6]
When helm is setup with tls the whatup plugin no longer works as it doesn't support the `--tls` flag.

```
$ helm whatup
Error: rpc error: code = Unavailable desc = transport is closing

$ helm --tls whatup
Error: unknown flag: --tls
```

**Comment by bacongobbler** [comment]
feel free to contribute fixes. I no longer use this plugin but I'm more than happy to continue reviewing PRs and cutting releases. :)

**Comment by l0nax** [comment]
@arthurk I've forked the Repository and added TLS Support: https://github.com/fabmation-gmbh/helm-whatup

**Comment by l0nax** [comment]
@arthurk  Feature has been implemented in Version [v0.2.0](https://github.com/fabmation-gmbh/helm-whatup/blob/v0.2.0/CHANGELOG.md) in https://github.com/fabmation-gmbh/helm-whatup/

**Comment by davidkarlsen** [comment]
@l0nax:
```
helm plugin remove whatup;helm plugin install https://github.com/fabmation-gmbh/helm-whatup
Error: Plugin: whatup not found
Downloading https://github.com/bacongobbler/helm-whatup/releases/download/v0.1.0/helm-whatup-0.1.0-darwin-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   627    0   627    0     0   1855      0 --:--:-- --:--:-- --:--:--  1860
100 4220k  100 4220k    0     0  2005k      0  0:00:02  0:00:02 --:--:-- 2815k
Preparing to install into /Users/et2448/.helm/plugins/helm-whatup
helm-whatup installed into /Users/et2448/.helm/plugins/helm-whatup/helm-whatup
check if installed charts are out of date (helm-whatup canary)

Usage:
  whatup [flags]

Flags:
  -h, --help   help for whatup
Installed plugin: whatup
```

It still downloads the bacongobbler version

**Comment by l0nax** [comment]
@davidkarlsen 
No it doesn't:

```bash
helm plugin install https://github.com/fabmation-gmbh/helm-whatup
Downloading https://github.com/fabmation-gmbh/helm-whatup/releases/download/v0.2.0/helm-whatup-0.2.0-linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   627    0   627    0     0   2668      0 --:--:-- --:--:-- --:--:--  2668
100 7979k  100 7979k    0     0  3234k      0  0:00:02  0:00:02 --:--:-- 3821k
Preparing to install into /root/.helm/plugins/helm-whatup
helm-whatup installed into /root/.helm/plugins/helm-whatup/helm-whatup
check if installed charts are out of date (helm-whatup canary)

Usage:
  whatup [flags]

Flags:
  -D, --deb             Print Debug Logs, defaults to false.
  -d, --devel           Whether to include pre-releases or not, defaults to false.
  -h, --help            help for whatup
  -o, --output string   Output format, choose from plain, json, yaml, table (default "table")
Installed plugin: whatup

```

Can you re-try it?

**Comment by davidkarlsen** [comment]
Yeah - same -  installs bacongobbler version

**Comment by davidkarlsen** [comment]
I'm on Client: &version.Version{SemVer:"v2.14.2", GitCommit:"a8b13cc5ab6a7dbef0a58f5061bcc7c0c61598e7", GitTreeState:"clean"}
fwiw

**Comment by l0nax** [comment]
I run the same Helm Version.

Can you please run the Install Command with the Debug Flag and post the Output?
I didn't manage to reproduce it.

```helm plugin --debug install https://github.com/fabmation-gmbh/helm-whatup```

**Comment by davidkarlsen** [comment]
```
helm plugin --debug install https://github.com/fabmation-gmbh/helm-whatup
[debug] updating https://github.com/fabmation-gmbh/helm-whatup
[debug] symlinking /Users/et2448/.helm/cache/plugins/https-github.com-fabmation-gmbh-helm-whatup to /Users/et2448/.helm/plugins/helm-whatup
[debug] loading plugin from /Users/et2448/.helm/plugins/helm-whatup
[debug] running install hook: &{/bin/sh [sh -c $HELM_PLUGIN_DIR/install-binary.sh] []  <nil> <nil> <nil> [] %!s(*syscall.SysProcAttr=<nil>) %!s(*os.Process=<nil>) <nil> <nil> <nil> %!s(bool=false) [] [] [] [] %!s(chan error=<nil>) %!s(chan struct {}=<nil>)}
Downloading https://github.com/bacongobbler/helm-whatup/releases/download/v0.1.0/helm-whatup-0.1.0-darwin-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   627    0   627    0     0   1904      0 --:--:-- --:--:-- --:--:--  1905
100 4220k  100 4220k    0     0  1193k      0  0:00:03  0:00:03 --:--:-- 1570k
Preparing to install into /Users/et2448/.helm/plugins/helm-whatup
helm-whatup installed into /Users/et2448/.helm/plugins/helm-whatup/helm-whatup
check if installed charts are out of date (helm-whatup canary)

Usage:
  whatup [flags]

Flags:
  -h, --help   help for whatup
Installed plugin: whatup
```

**Comment by l0nax** [comment]
- Delete the Directory `$HOME/.helm/plugins/helm-whatup`
or
- Go into `$HOME/.helm/plugins/helm-whatup` and run `git pull && git checkout master`

And re-run `helm install [...]`

It looks like that Helm has cached the Plugin and in previous Commits the `install-binary.sh` was not edited.

**Comment by davidkarlsen** [comment]
Found out -  need to delete the cache/plugins dir!

**Comment by l0nax** [comment]
@bacongobbler Can this Issue be closed?
TLS Support was implemented in [v0.2.0](https://github.com/fabmation-gmbh/helm-whatup/releases/tag/v0.2.0)

**Comment by davidkarlsen** [comment]
@l0nax @bacongobbler shouldn't it be merged upstream (e.g. this repo) for this issue to close? Or are you abondoning it @bacongobbler so the fork should live on?

**Comment by l0nax** [comment]
@davidkarlsen My fork is abandoning @bacongobbler Repo.

Because I can better maintain my Fork instead of sending every time a PR.

**Comment by bacongobbler** [comment]
I would prefer if you would contribute the PR upstream. As mentioned before, I am happy to accept patches and cut releases as necessary. However, I understand if you feel the need to maintain your own fork.

So I'd rather keep this issue open until TLS support is implemented here.

**Comment by bacongobbler** [comment]
I had a look at @l0nax's implementation and saw that it was missing several core asks, like the `--tls` flag originally asked by the OP. The code was also quite shoddy, so I had to go and rewrite the implementation entirely from scratch.

I also noticed that table support was added there, so I implemented that here as well. There should be no difference in features compared to the upstream fork as to what's in here.

TLS support was implemented in https://github.com/bacongobbler/helm-whatup/commit/c89ebc663a9366be9de09d9bea68fcab6f0e44ea. Will cut an 0.2.0 today.
